Library/Homebrew/utils/curl.rb

```diff
@@ -238,6 +238,48 @@ def curl_check_http_content(url, specs: {}, user_agents: [:default], check_conte
       "The URL #{url} may be able to use HTTPS rather than HTTP. Please verify it in a browser."
     end
 
+    # Separates cURL response text into headers and body content. Headers (if
+    # any) are split into an array, where each member is a string of headers
+    # for an individual response.
+    # @param response_text [String] The response text from `curl` containing
+    #   headers, body, or both
+    # @return [Array<Array<String>, String, nil>] An array containing an array
+    #   of headers and the body content.
+    def response_headers_and_body(response_text)
+      headers = []
+      while response_text.match?(%r{\AHTTP/.* (\d+)})
```

Comment:
While loops make me tetchy. I'd be good to swap to a loop that cannot be unbounded.

Reply:
I generally agree but it may be worth mentioning that this is basically just a modification of part of the existing approach that was in `curl_http_content_headers_and_checksum`:

```ruby
status_code = :unknown
while status_code == :unknown || status_code.to_s.start_with?("3")
  headers, _, output = output.partition("\r\n\r\n")
  status_code = headers[%r{HTTP/.* (\d+)}, 1]
  location = headers[/^Location:\s*(.*)$/i, 1]
  final_url = location.chomp if location
end
```

The main difference is that `response_headers_and_body` is more general purpose and the parts that work with the headers are separated into `response_status_code_and_location` (which works for `curl --head` requests as well).

I vaguely remember trying to replace this `while` loop when I was originally working on this but the approaches I tried ended up introducing bugs. Did you have any particular alternative in mind?